.github/workflows/samples/app.yaml

```diff
@@ -0,0 +1,55 @@
+name: Application
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "README.md"
+      - "renovate.json"
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - "README.md"
+      - "renovate.json"
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+  pull-requests: write
+
+jobs:
+  docker_build_artifact:
+    if: github.event_name == 'pull_request'
+    uses: ./.github/workflows/docker-build.yaml
+      
+  docker_push_ecr:
+    if: github.event_name == 'push'
+    strategy:
+      fail-fast: false
+      matrix:
+        environment: [app-dev, app-int, app-prod]
```

Comment:
Can we add a stack into the matrix
```
matrix:
  stack: app
  environment: [dev, int, prod]

with:
  environment: ${{ matrix.stack }}-${{ matrix.environment }}
```
We get get the best of both worlds then.

Reply:
That's a good idea ! ❤️

Reply:
I updated both infra and dev